doc/west/repo-tool.rst

```diff
@@ -0,0 +1,266 @@
+  ``BRANCH_NAME`` in each of the specified projects (default: all projects).
+
+  Like all branches created by the multi-repo commands, the newly created
+  branches track ``manifest-rev``.
+
+  If no arguments are passed to ``west branch``, local branches from all
+  projects are listed, along with the projects they appear in.
+
+- ``west checkout [-b] BRANCH_NAME [PROJECT ...]]``: Checks out the branch
+  ``BRANCH_NAME`` in each of the specified projects (default: all projects).
+  This command is a no-op for projects that do not have the specified branch.
+
+  With ``-b``, creates and checks out the branch if it does not exist in a
+  project (like ``git checkout -b``), instead of ignoring the project.
+
+Miscellanous commands
+*********************
+
+These commands perform miscellaneous functions.
+
+- ``west list``: Lists project information from the manifest (URL, revision,
+  path, etc.), along with other manifest-related information.
+
+- ``west diff [PROJECT ...] [ARGUMENT ...]``: Runs a multi-repo ``git diff``
+  for the specified projects (default: all projects).
+
+  Extra arguments are passed as-is to ``git diff``.
+
+- ``west status [PROJECT ...] [ARGUMENT ...]``: Like ``west diff``, for
+  running ``git status``.
+
+- ``west update [--update-west] [--update-manifest]``: Updates the west and
+  manifest repositories.
+
+  Normally, the west and manifest repositories are updated automatically
+  whenever a command that fetches upstream data is run. This self-update is
+  skipped (with a warning) if not possible via fast-forward (i.e., if you've
+  made modifications to the west or manifest repositories).
+
+Example workflow
+================
+
+This section gives an example workflow that clones the Zephyr repositories,
+creates and switches between two work branches, updates (rebases) a work branch
+to the latest manifest revision, and inspects upstream changes.
+
+1. First, we initialize West with the default Zephyr manifest, and fetch all
+   projects:
+
+   .. code-block:: console
+
+      $ west init
+      $ west fetch
+
+   .. note::
+
+      ``west clone`` could be used instead of ``west fetch``, to automatically
+      create and check out a branch in each project, named after the project's
+      revision. In this example, we create all branches ourselves.
+
+2. Next, we create and check out a branch ``xy-feature`` for implementing a
+   feature that spans the ``proj-x`` and ``proj-y`` projects:
+
+   .. code-block:: console
+
+      $ west checkout -b xy-feature proj-x proj-y
+
+   This creates a Git branch named ``xy-feature`` in ``proj-x`` and ``proj-y``.
+   The new branches are automatically set up to track the ``manifest-rev``
+   branch in their respective repositories.
+
+   We work on the repositories as usual.
+
+3. While working on the feature, an urgent bug comes up that spans all
+   projects. We commit/stash the partially implemented feature and check out a
```

Comment:
Hrm, west checkout doesn't actually stash the working tree; I think this could be cleared up a bit.

Reply:
There was an implied "ourselves" in there. I rephrased it a bit. :)